### PR TITLE
Added unit tests to the cli/command/idresolver package

### PR DESCRIPTION
**Pull request by adshmh**
This PR adds unit tests to the cli/command/idresolver package. Part of the work on #31217

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

**- What I did**
Added unit tests to the cli/command/idresolver package

**- How I did it**
Added unit tests and supporting code (builders) for cli/command/idresolver package 

**- How to verify it**
Run the unit tests.

- There are no unit tests for the cli/command/idresolve package in current version (no test files)

- With this PR:
    $ make test-unit
    ....
    ok  	github.com/docker/docker/cli/command/idresolver	0.023s	coverage: 95.8% of statements
    ....

**- Description for the changelog**
Added unit tests to cli/command/idresolver package.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/command/idresolver/idresolver_test.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+	idResolver := New(cli.Client(), false)
+	_, err := idResolver.Resolve(context.Background(), struct{}{}, "nodeID")
+
+	assert.Error(t, err, "unsupported type")
+}
+
+func TestResolveWithNoResolveOption(t *testing.T) {
+	resolved := false
+	cli := test.NewFakeCli(&fakeClient{
```
Comment (dnephin) [inline cli/command/idresolver/idresolver_test.go]: This `FakeCli` isn't really being used, right? You can just create a client and pass the client in directly.

I believe the same is true for the other tests as well.

Comment (adshmh) [inline cli/command/idresolver/idresolver_test.go]: Thank you for the review. I updated the PR to create and directly pass the client.

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

**Review by dnephin — APPROVED** [review]
LGTM

**Pull request merged**